dist/install.sh

```diff
@@ -117,11 +125,18 @@ esac
 TARBALL_URL="https://github.com/pulumi/pulumi/releases/download/v${VERSION}/"
 TARBALL_URL_FALLBACK="https://get.pulumi.com/releases/sdk/"
 TARBALL_PATH=pulumi-v${VERSION}-${OS}-${ARCH}.tar.gz
+PULUMI_INSTALL_ROOT=${INSTALL_ROOT}
 
-if ! command -v pulumi >/dev/null; then
+if [ "$PULUMI_INSTALL_ROOT" = "" ]; then
+    # Default to ~/.pulumi
+    PULUMI_INSTALL_ROOT="${HOME}/.pulumi"
+fi
+PULUMI_CLI="${PULUMI_INSTALL_ROOT}/bin/pulumi"
+
+if [ ! -e "${PULUMI_CLI}" ]; then
```

Comment:
```suggestion
if [ -d "${PULUMI_CLI}" ]; then
    say_red "The given root is a folder, refusing to proceed"
    exit 1
elif [ ! -f "${PULUMI_CLI}" ]; then
```

Otherwise we could potentially fail down the line somewhere.  Probably not a big issue, and probably unlikely to come up in practice, but I think this would make it a slightly nicer user experience.